netbox/dcim/api/views.py

```diff
@@ -414,6 +415,25 @@ def get_serializer_class(self):
 
         return serializers.DeviceWithConfigContextSerializer
 
+    @transaction.atomic
+    def create(self, request, *args, **kwargs):
```

Comment:
IMO we should move this to a mixin class in case we need it for other views in the future.